pkg/apis/servicecatalog/v1alpha1/types.go

```diff
@@ -57,15 +57,44 @@ type BrokerSpec struct {
 // BrokerAuthInfo is a union type that contains information on one of the authentication methods
 // the the service catalog and brokers may support, according to the OpenServiceBroker API
 // specification (https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md).
-//
-// Note that we currently restrict a single broker to have only one of these fields
-// set on it.
 type BrokerAuthInfo struct {
-	// BasicAuthSecret is a reference to a Secret containing auth information the
-	// catalog should use to authenticate to this Broker using basic auth.
-	BasicAuthSecret *v1.ObjectReference `json:"basicAuthSecret,omitempty"`
+	// Basic provides configuration for basic authentication
+	Basic *BasicAuthConfig `json:"basic,omitempty"`
+	// Bearer provides configuration for bearer token authentication
+	Bearer *BearerTokenAuthConfig `json:"bearer,omitempty"`
```

Comment:
Could we not call this BearerTokenAuthConfig and call it something like:
AuthenticationHeader

That way it could be used for something else in the future. If I understand this PR correctly, refresh -> bearer token happens offline somewhere else? So, would be good to at least walk through and see what that process looks like. Is it assumed that offline process handles the conversion [aka, possibly continously refreshing [unnecessarily if no calls to broker are made]]. Is there no plan to store the refresh token as part of the Service Catalog, and having a way to convert that into a bearer token?